python/adjoint/filters.py

```diff
@@ -795,6 +805,9 @@ def smoothed_projection(
     # The distance for the center of the pixel to the nearest interface
     d = (eta - rho_filtered) / rho_filtered_grad_norm_eff
 
+    # Account for an erosion or dilation
+    d = d + erosion_dilation
```

Comment:
@stevengj something as simple as this, correct?

unfortunately, we still have the same problem I mention above. At this point, the routine is just falling back to the naive projection everywhere:

<img width="992" height="341" alt="image" src="https://github.com/user-attachments/assets/93245070-bf08-4c1d-a058-b79db1b9ff5b" />

Note that $\beta=\infty$ for this test.

Reply:
I think the problem is with `needs_smoothing`.

If I just look at $F$ (and ignore the where clause) I get the expected perturbation. (Note that I have to clip the min and max because things get ugly when the spatial gradient breaks down in uniform regions.)

<img width="1006" height="351" alt="image" src="https://github.com/user-attachments/assets/61fab86f-ba5d-4bb2-9ac0-6b6e451e86da" />